packages/toolpad-app/src/components/Home.tsx

```diff
@@ -112,45 +120,178 @@ function AppDeleteDialog({ app, onClose }: AppDeleteDialogProps) {
-        <CardContent>
-          <Typography gutterBottom variant="h5" component="div">
-            {app ? app.name : <Skeleton />}
-          </Typography>
-          <Typography variant="body2" color="text.secondary">
-            {app ? `Created: ${app.createdAt.toLocaleDateString('short')}` : <Skeleton />}
-          </Typography>
-        </CardContent>
-      </CardActionArea>
-      <CardActions>
-        <Button
-          size="small"
-          component="a"
-          href={app ? `/_toolpad/app/${app.id}/editor` : ''}
-          disabled={!app}
-        >
-          Edit
-        </Button>
-        <Button size="small" disabled={!app} onClick={onDelete}>
-          delete
-        </Button>
-      </CardActions>
-    </Card>
+    <React.Fragment>
+      <Card sx={{ gridColumn: 'span 1' }}>
+        <CardHeader
+          action={
+            <IconButton
+              aria-label="settings"
+              aria-controls={menuOpen ? 'basic-menu' : undefined}
+              aria-haspopup="true"
+              aria-expanded={menuOpen ? 'true' : undefined}
+              onClick={handleMenuClick}
+            >
+              <MoreVertIcon />
+            </IconButton>
+          }
+          disableTypography
+          title={
+            editingTitle ? (
+              <TextField
+                variant="standard"
+                size="small"
+                inputRef={appTitleInput}
+                sx={{ paddingBottom: '4px' }}
+                InputProps={{ sx: { fontSize: '1.5rem', height: '1.5em' } }}
+                onKeyUp={handleAppTitleInput}
+                onBlur={handleAppTitleBlur}
+                defaultValue={appTitle}
+              />
+            ) : (
+              <Typography gutterBottom variant="h5" component="div">
+                {app ? appTitle : <Skeleton />}
+              </Typography>
+            )
+          }
+          subheader={
+            <Typography variant="body2" color="text.secondary">
+              {app ? `Created: ${app.createdAt.toLocaleDateString('short')}` : <Skeleton />}
+            </Typography>
+          }
+        />
+
+        <CardActions>
+          <Button
+            size="small"
+            component="a"
+            href={app ? `/_toolpad/app/${app.id}/editor` : ''}
+            disabled={!app}
+          >
+            Edit
+          </Button>
+          <Button size="small" component="a" href={app ? `deploy/${app.id}` : ''} disabled={!app}>
+            View
+          </Button>
+        </CardActions>
+      </Card>
+      <Menu
+        id="basic-menu"
+        anchorEl={menuAnchorEl}
+        open={menuOpen}
+        onClose={handleMenuClose}
+        MenuListProps={{
+          'aria-labelledby': 'basic-button',
+          dense: true,
+        }}
+      >
+        <MenuItem onClick={handleRenameClick}>
+          <DriveFileRenameOutlineIcon sx={AppCardMenuIconStyles} />
```

Comment:
Perhaps we can do it like in the demo? https://mui.com/material-ui/react-menu/#icon-menu